content/en/blog/2022-08-05-effective-programmers-read-code.md

```diff
@@ -0,0 +1,59 @@
+
+The glass castle is essential for how I write good code and recently, I've started to pay more attention to how I construct the glass castle. When I sit down to implement a feature or make a change, the very first thing I do is identify candidate locations in the code where the change needs to be made. These locations are identified in a top down fashion looking at the organization of the code and quickly scanning through package and module names and package documentation.
+
+Once I find the candidate locations, I continue in a bottom up fashion identifying the functional dependencies surrounding the location and putting together components and interactions tagged with the input and output data contracts that they make. Often, I have to read the code of the external dependencies of the project to figure out the best possible solutions. I continue grouping, linking, and encapsulating code in my head until I feel like I have enough of a picture that I'm confident in the change I'm about to make.
+
+Before I've written even a single line of code, I've read hundreds of lines of code. The more complex the change or the system (or the longer I was away from the system before making a change), the more time I have to spend reading code before writing it. That's when I realized that as a programmer I spend far more time reading code than writing it.[^3]
+
+My students often ask me how to become a better programmer and my answer was always "read code, particularly good code, and you'll become a better programmer". The theory was similar to writing blogs or books -- the more you read, the better you write and there is tons of excellent open source code. More importantly, I noticed that my students would ignore blocks of code to quickly get to writing code and more often than not, the ignored blocks were where the problems came from. What I understand now is that it's not just practice or a way to become better or avoid bugs. Instead, reading code is the most essential part of our job as programmers.
```

Comment:
Suggestions: "...and my answer has always been "read code..."

"....the more you read, the better you'll write - and there are tons of resources for excellent open source code."

"...get to writing code and, more often than not, those ignored blocks were where the problems came from."

"it's not just practice, or a way to become better or avoid bugs."